main.js

```diff
@@ -257,3 +253,57 @@ function winnerNameOutput() {
     winnerName.innerText = twoName.value;
   }
 }
+
+// function below will reset game to default with new generated correct number
+document.addEventListener('keyup', enableReset);
+resetButton.addEventListener('click', resetGame);
+
+function enableReset() {
+  var hasMin = minRange.value !== "";
+  var hasMax = maxRange.value !== "";
+  var hasOneName = oneName.value !== "";
+  var hasTwoName = twoName.value !== "";
+  var hasOneGuess = oneGuess.value !== "";
+  var hasTwoGuess = twoGuess.value !== "";
+  var isFilled = hasMin || hasMax || hasOneName || hasTwoName || hasOneGuess || hasTwoGuess;
+  if (isFilled) {
+    resetButton.disabled = false;
+  } else {
+    resetButton.disabled = true;
+  }
+}
+
+function resetGame() {
+  defaultSetRange();
+  clearAllField();
+  defaultLastetGuess();
+  submitButton.disabled = true;
+  clearButton.disabled = true;
+  updateButton.disabled = true;
+  resetButton.disabled = true;
+}
+
+function clearAllField() {
+  oneName.value = "";
+  twoName.value = "";
+  oneGuess.value = "";
+  twoGuess.value = "";
+  minRange.value = "";
+  maxRange.value = "";
+}
+function defaultSetRange() {
+  currentMin = 1;
+  currentMax = 100;
+  correctNumber = getRandomRange(currentMin, currentMax);
+  minSpan.innerText = "1";
+  maxSpan.innerText = "100";
+}
```

Comment:
I think this function is redundant with the updateRange function we already have

Reply:
I don't think that the updateRange restore the Number Guesser values to default.  When reset button is click, all of those elements in defaultSetRange will restore to default value.  updateRange do not restore default value because updateRange will store values based on user input, not default value.

Reply:
You're right! Maybe we can change that later so that they both work that way.